### ISSUE TITLE
v2.9.1 is sometimes unable to suppress CA1031 in GlobalSuppressions.cs

### ISSUE THREAD

**Issue by Andrew-Webb** [dotnet/roslyn-analyzers#2268]
#### Analyzer package

[Microsoft.CodeAnalysis.FxCopAnalyzers](https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers)

#### Package Version

v2.9.1

#### Diagnostic ID

[CA1031](https://docs.microsoft.com/en-us/visualstudio/code-quality/ca1031-do-not-catch-general-exception-types?view=vs-2017)

#### Repro steps

1. Upgrade from v2.6.3 NuGet package of FxCopAnalyzers to v2.9.1
1. Rebuild solution
1. Attempt to suppress multiple instances of CA1031 in suppression file.

#### Expected behavior

All suppressions actually work.

#### Actual behavior

Some suppressions don't work.  I.e. the warning is still reported.

In my code I have:-
```
catch (Exception ex)
{
    throw InternalServiceException("blah bah some message " + ex.Message, ex);
}
```
in numerous places.  I.e. I am converting System.Exception to another exception type.  For the time being at least, I want this behaviour to remain as-is.

I suppressed and suppressed and suppressed in suppression file over and over and over before I realized that some suppressions just didn't work, and the same instances were being reported as warnings... forever!  I ended up with a suppressions file with the exact same lines repeated over and over.

In the end I deleted the numerous entries from the suppressions file, downgraded to v2.6.3 and opened this issue.

**Comment by dkalbertson** [comment]
I'm seeing the same issue in 2.9.3

**Comment by mavasani** [comment]
@Andrew-Webb @dkalbertson I am unable to repro this issue with 2.9.3:

1. Create a C# .NetStandard library with following source code:
```csharp
using System;

public class MyException : Exception
{
    void M()
    {
        try
        {
        }
        catch (Exception ex)
        {
            throw CreateException(ex);
        }
    }

    MyException CreateException(Exception ex)
    {
        return new MyException();
    }
}
```
2. Add NuGet package reference: https://www.nuget.org/packages/Microsoft.CodeAnalysis.FxCopAnalyzers/2.9.3
3. Build project and verify no CA1031
4. Comment out the line `throw CreateException(ex);` and verify CA1031 is raised.
5. Use the suppression code fix to add a suppression in separate suppression file:
```csharp
[assembly: System.Diagnostics.CodeAnalysis.SuppressMessage("Design", "CA1031:Do not catch general exception types", Justification = "<Pending>", Scope = "member", Target = "~M:MyException.M")]
```
6. Verify no CA1031 is fired.
7. Uncomment the line commented in step 4. Verify no CA1031 is fired.

**Comment by dkalbertson** [comment]
@Andrew-Webb @mavasani After digging in, it appears my issue was due to the offending method returning a ValueTuple and it was not added to the global suppression file properly (the return type was left as "~System."

**Comment by mavasani** [comment]
@dkalbertson Thanks.

I think @Andrew-Webb issue is a dupe of https://github.com/dotnet/roslyn-analyzers/issues/2278, which was fixed in 2.9.3. I am going to resolve this bug as a duplicate. Please re-open if you still see it.
